### PR TITLE
Feat payment

### PR DESCRIPTION
**Pull request by MertsAnastasiya**

### CODE REVIEW

File: src/modules/app/app.ts [hunk 1]
```diff
@@ -205,6 +205,17 @@ export class App {
                 this.searchParams.clearUrl();
                 this.globalFiltres.clearFilters();
             }
+            case 'buy': {
+                const paymentForm = new PaymentForm(MAIN_CONTAINER, this.onButtonClick);
+                paymentForm.drawForm();
```
Comment (alexey-ulashchick) [inline src/modules/app/app.ts:210]:
```suggestion
                new PaymentForm(MAIN_CONTAINER, this.onButtonClick).paymentForm.drawForm();
```

File: src/modules/app/app.ts [hunk 2]
```diff
@@ -205,6 +205,17 @@ export class App {
...
+                break;
+            }
+            case 'pay': {
+                console.log('pay');
+                document.querySelector('.modal-window')!.innerHTML = `<p class="message">The order accepted!</p>`;
+                setTimeout(() => window.location.href = window.location.origin, 3000);
```
Comment (alexey-ulashchick) [inline src/modules/app/app.ts:216]: Careful here. User might navigate (change location) within this 3 seconds interval. Then callback execution will be looking weird.

File: src/modules/paymentForm.ts [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        cardNumber.value = value
+            .replace(/[^\d]/g, '')
+            .replace(/(.{4})/g, '$1 ')
+            .trim();
+
+        if (value.length === MAX_LENGTH_CARD_NUMBER) {
+            return true;
+        } else {
+            return false;
+        }
```
Comment (alexey-ulashchick) [inline src/modules/paymentForm.ts:79]: =)
```suggestion
        return value.length === MAX_LENGTH_CARD_NUMBER;
```

File: src/modules/paymentForm.ts [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+        const fullName = document.querySelector(
+            '.input_name'
+        )! as HTMLInputElement;
+        const value = fullName.value;
+        const regexp: RegExp = /^[A-Za-z]{3,}\s[A-Za-z]{3,}(\s?[A-Za-z]{3,})*$/;
+        if (regexp.test(value)) {
+            return true;
+        } else {
+            return false;
+        }
```
Comment (alexey-ulashchick) [inline src/modules/paymentForm.ts:109]: ????
Cab we just return regexp.test(value)?

File: src/modules/paymentForm.ts [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+    private checkCvv(): boolean {
+        const cvv = document.querySelector('.input_code')! as HTMLInputElement;
+        cvv.maxLength = MAX_LENGTH_CVV;
+        const value = cvv.value;
+        cvv.value = value.replace(/[^\d]/g, '');
+        if (value.length === MAX_LENGTH_CVV) {
+            return true;
+        } else {
+            return false;
+        }
```
Comment (alexey-ulashchick) [inline src/modules/paymentForm.ts:121]: same

**Review by alexey-ulashchick — CHANGES_REQUESTED** [review]

File: src/modules/app/app.ts (hunk 1 above)
Comment (MertsAnastasiya) [reply src/modules/app/app.ts:210]: Fixed

File: src/modules/paymentForm.ts (hunk 3 above)
Comment (MertsAnastasiya) [reply src/modules/paymentForm.ts:79]: Fixed)

File: src/modules/paymentForm.ts (hunk 4 above)
Comment (MertsAnastasiya) [reply src/modules/paymentForm.ts:109]: Fixed

File: src/modules/paymentForm.ts (hunk 5 above)
Comment (MertsAnastasiya) [reply src/modules/paymentForm.ts:121]: Fixed.

File: src/modules/app/app.ts (hunk 2 above)
Comment (MertsAnastasiya) [reply src/modules/app/app.ts]: Fixed

**Review by alexey-ulashchick — APPROVED** [review]

**Pull request merged**